### PR TITLE
feat: add unit test for building block

### PR DESCRIPTION
**Pull request by jacquesikot**
## Description
Add unit tests for the building block card component

`BuildingBlock`
    ✓ renders building block details correctly
    ✓ navigates to the template URL when clicked
    ✓ triggers onForkTemplateClick when the fork button is clicked
    ✓ does not trigger onForkTemplateClick when the button is hidden
    ✓ opens the fork modal when the fork button is clicked
#### PR fixes following issue(s)
Fixes # (issue number)
#### Type of change
- New feature (non-breaking change which adds functionality)

- [ ] Manual
- [ ] JUnit
- [x] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **New Features**
  - Introduced a unit test suite for the `BuildingBlock` component to ensure functionality and stability.
  
- **Refactor**
  - Updated the `ForkTemplate` component to improve testability and consistency in the DOM structure.

- **Tests**
  - Added tests for user interactions within the `BuildingBlock` component, such as navigation and modal interactions.

### CODE REVIEW

File: app/client/src/pages/Templates/BuildingBlock/BuildingBlock.test.tsx [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    "https://mockapp.com/screenshot1.jpg",
+    "https://mockapp.com/screenshot2.jpg",
+  ],
+  widgets: [],
+  functions: ["Function1", "Function2"],
+  useCases: ["UseCase1", "UseCase2"],
+  datasources: ["Datasource1", "Datasource2"],
+  pages: [],
+  allowPageImport: true,
+};
```
Comment (rahulbarwal) [inline app/client/src/pages/Templates/BuildingBlock/BuildingBlock.test.tsx:47]: This seems like a data for this test. We should put this in a separate `.json` file?

**Review by rahulbarwal — DISMISSED** [review]
Looks good, see if the comment makes sense.

File: app/client/src/pages/Templates/BuildingBlock/BuildingBlock.test.tsx (hunk 1 above)
Comment (jacquesikot) [reply app/client/src/pages/Templates/BuildingBlock/BuildingBlock.test.tsx]: I have moved this to an existing constants file within the templates folder.

**Review by rahulbarwal — CHANGES_REQUESTED** [review]
Test constant and code constants should not be mixed. 
Lets keep them seperate(similar to fixtures in cypress).

**Review by rahulbarwal — APPROVED** [review]

**Pull request merged**